lib/handler/pool/state.ex

```diff
@@ -0,0 +1,94 @@
+defmodule Handler.Pool.State do
+  @moduledoc false
+  alias __MODULE__
+  alias Handler.Pool.{InsufficientMemory, NoWorkersAvailable}
+
+  defstruct running_workers: 0,
+            bytes_committed: 0,
+            workers: %{},
+            pool: nil
+
+  @type t :: %State{
```

Comment:
This typespec here is super useful 👌🏼 I've referred to it twice now just reading this PR